src/AzureExtension/DevBox/DevBoxInstance.cs

```diff
@@ -684,9 +689,11 @@ public IAsyncOperation<ComputeSystemOperationResult> DoPinActionAsync(string loc
             var psi = new ProcessStartInfo();
             psi.UseShellExecute = true;
             psi.FileName = string.Format(CultureInfo.InvariantCulture, ProtocolPinString, location, pinAction, WorkspaceId, DisplayName, Environment, Username);
+
             Process? p = Process.Start(psi);
             if (p != null)
             {
+                SetForegroundWindow(p.MainWindowHandle);

```

Comment:
I think there are a couple of races here:

1. The process may exit by the time SetForegroundWindow is called.
2. Main window may not exist yet.

So, the results may vary in the wild.
According to [docs](https://learn.microsoft.com/en-us/dotnet/api/system.diagnostics.process.mainwindowhandle?view=net-8.0#system-diagnostics-process-mainwindowhandle:~:text=InvalidOperationException,process%20has%20exited.), accessing MainWindowHandle can throw InvalidOperationException if the process is already exited, so you may want to handle exceptions here.

Reply:
Good points.  I've switched to AllowSetForegroundWindow which uses a PID so now #2 is irrelevant. And I've added an Event for synchronization so that WindowsApp will not proceed until the Event signals that DevHomeAzureExtensions has given WindowsApp foreground rights.  And now that I'm not using MainWindowHandle I don't have to worry about that exception (although it wouldn't have been a problem because exceptions get handled higher up in the callstack).